### ISSUE TITLE
Enable Prow plugins for etcd PRs

### ISSUE THREAD

**Issue by ArkaSaha30** [etcd-io/etcd#18110]
### What would you like to be added?

Plugin enablement for etcd in [test-infra](https://github.com/kubernetes/test-infra/blob/master/config/prow/plugins.yaml). 
Suggested plugins to be added:
- [x] cherrypicker: https://prow.k8s.io/command-help#cherrypick
- [x] size: https://github.com/kubernetes/test-infra/blob/master/config/prow/plugins.yaml#L145-L151
- [x] needs-rebase: https://github.com/kubernetes/test-infra/blob/master/config/prow/plugins.yaml#L1661-L1664
- [x] lgtm: https://prow.k8s.io/command-help#lgtm
- [x] approve: https://prow.k8s.io/command-help#approve
- [x] hold https://prow.k8s.io/command-help#hold
- [x] lifecycle https://prow.k8s.io/command-help#lifecycle

### Why is this needed?

There are useful Prow plugins that etcd can leverage to minimize manual efforts. One of them is the cherrypicker plugin, which can backport any PR to a specific branch with just a `/cherry-pick <branch_name>` comment on the PR. 
Example: https://github.com/containerd/containerd/pull/10257#issuecomment-2132261738

**Comment by ArkaSaha30** [comment]
cc @ahrtr @jmhbnz

**Comment by ahrtr** [comment]
Sounds good to me.  `/cherry-pick` might work for 3.5, but not for 3.4, which has huge difference with main.

**Comment by jmhbnz** [comment]
Thanks for proposing this @ArkaSaha30. Enabling the cherry pick plugin makes sense to me, please feel free to raise a pr if you have capacity 🙏🏻

**Comment by jmhbnz** [comment]
Hi @ArkaSaha30 - We raised this at sig-etcd triage meeting, are there any other plugins you suggest we enable or can we close this now?

**Comment by ivanvc** [comment]
Based on https://github.com/kubernetes/k8s.io/issues/6102#issue-2000912419, do we also want to enable tide, lgtm, and approve?

**Comment by ArkaSaha30** [comment]
> Based on https://github.com/kubernetes/k8s.io/issues/6102#issue-2000912419, do we also want to enable tide, lgtm, and approve?

Yes! Along with these we can also add 'size' which will label the PRs based on how lengthy it is. WDYT @jmhbnz ?

**Comment by jmhbnz** [comment]
> Based on [kubernetes/k8s.io#6102 (comment)](https://github.com/kubernetes/k8s.io/issues/6102#issue-2000912419), do we also want to enable tide, lgtm, and approve?

I am not confident we are ready as a project for `tide` / auto merging of pull requests.  It would be helpful to update this issue with a description of what the proposed plugin does and what impact this would have for the etcd project.  

I am happy to enable the pr sizing plugin as that is very low risk.

**Comment by jmhbnz** [comment]
Discussed during sig-etcd triage meeting, the next major step for this issue is to enable `tide` for automatically merging pull requests.

This will be a significant change for `etcd-io/etcd` so we need to discuss this more broadly and ensure the community is ready, next steps would be:
- Discuss at a community meeting - Added agenda item
- Raise a #sig-etcd slack discussion
- Add a detailed comment here with the behavior for tide
- Draft a pull request with a /hold

**Comment by ivanvc** [comment]
I think we should also enable the hold plugin. Reference: https://github.com/etcd-io/etcd/pull/19810#issuecomment-2832338489

It may be a good candidate for a good first issue.

**Comment by jmhbnz** [comment]
> I think we should also enable the hold plugin. Reference: https://github.com/etcd-io/etcd/pull/19810#issuecomment-2832338489

> It may be a good candidate for a good first issue.

Completely agree - I've edit this issue list to add it and added the labels.

**Comment by wendy-ha18** [comment]
I'm so sorry for tinkering around, I saw etcd using DCO (Developer Certificate of Origin) https://github.com/etcd-io/etcd/blob/main/DCO) 11 years ago , but since etcd is now a subproject of Kubernetes and [Kubernetes can only accept original source code from CLA signatories](https://github.com/kubernetes/community/blob/master/CLA.md), do you think in the future etcd-io should have `cla` plugin to enforce PRs submitted by contributors who have not signed the CLA cannot be merged? See [Sign the CLA](https://kubernetes.io/docs/contribute/new-content/#sign-the-cla) for more information.

**Comment by ivanvc** [comment]
Hi @wendy-ha18, @jmhbnz did some exploration on enabling the tide plugin for the etcd-io organization. One of the blockers was DCO vs. CLA. I don't recall if we documented this exploration somewhere. If not, maybe James can shed some light here.

I think @serathius opted to use DCO for its simplicity. Although it may be good to transition to CLA, it may be a more significant/lengthier project/conversation.

**Comment by wendy-ha18** [comment]
> One of the blockers was DCO vs. CLA

Is it because tide config required `cncf-cla: yes` label: https://github.com/kubernetes/test-infra/blob/master/config/prow/config.yaml#L701C7-L701C22 ? 

I just try to think out loud, if we want to use CLA, IMHO the process would be:
- Summit a [form](https://jira.linuxfoundation.org/plugins/servlet/desk/portal/4/create/143) to LFX to enable EasyCLA to etcd-io org ([read more](https://docs.linuxfoundation.org/lfx/easycla/v2-current/getting-started/easycla-faqs))=> only challenge here maybe is waiting time to get approve (?)
-  [Deploying Prow instance](https://docs.prow.k8s.io/docs/getting-started-deploy/) => done cause I saw `/ok-to-test` command, assuming Prow has been deployed, but feel free to correct me if  I'm wrong.
- [Configure Tide](https://docs.prow.k8s.io/docs/components/core/tide/config/)

P/S: I'm not very sure about the impact of moving DCO to CLA and using tide for auto merging PR in the wider etcd community, but a few folks did it before here is this old issue https://github.com/kubernetes/test-infra/issues/8286 if anyone want to reach out and ask about their experience.

**Comment by wendy-ha18** [comment]
A few prior dicussions from K8s steering committes and other contributors for references, but no conclusion has been made. Now I know why this topic is legacy conversation 🥲
https://github.com/kubernetes/community/issues/2649
https://github.com/kubernetes/steering/issues/74
https://kubernetes.slack.com/archives/C1TU9EB9S/p1682026728590429?thread_ts=1682025037.524659&cid=C1TU9EB9S
https://kubernetes.slack.com/archives/C1TU9EB9S/p1712668509882039

**Comment by joshjms** [comment]
Hi @Dhairya-Dudhatra , yes that's correct. You can refer to this https://github.com/kubernetes/test-infra/pull/32725 for reference.

**Comment by Dhairya-Dudhatra** [comment]
@lavishpal I was working on this issue but I see that you have already created a PR, in that case, go ahead take on this issue.

**Comment by ivanvc** [comment]
@Dhairya-Dudhatra, I'm sorry you couldn't finish working on the issue because someone else took it. I hope we can find another good first issue so you can start contributing to the project.

Thanks for picking up the issue, @lavishpal. I suggest giving the original contributor more time to answer (i.e., we usually wait for a week). Then, please assign the issue to yourself if you want to work on it.

**Comment by lavishpal** [comment]
Thanks for the clarification, @ivanvc .
Apologies for jumping in early , I did ping @Dhairya-Dudhatra and waited around 24 hours before proceeding since there was no response. I’ll make sure to follow the one-week guideline moving forward. Appreciate your understanding!

**Comment by jmhbnz** [comment]
Discussed during sig-etcd triage - one further plugin we should explicitly enable is `lifecycle`.

Adding /help-wanted

**Comment by lavishpal** [comment]
Hi @jmhbnz , Can i work on this issue?

**Comment by ivanvc** [comment]
For sure, @lavishpal.

/assign @lavishpal

**Comment by ivanvc** [comment]
Link to #19587

**Comment by nicomwong** [comment]
Hello, is this issue still relevant? I would like to contribute if I am not assigned to https://github.com/etcd-io/etcd/issues/19706. @ArkaSaha30

**Comment by ivanvc** [comment]
Hi @nicomwong, thanks for the interest. This is still valid as we want to enable tide. However, there are some technical limitations, which are not good first issues. We're keeping this issue open as a reminder that we still have some research work to do. Thanks again :)

**Comment by tysoncung** [comment]
I'm interested in working on this issue. Could you provide more context about what you're looking for? Any additional details about requirements or constraints would be helpful.

**Comment by jberkus** [comment]
Hey folks, since the whole original list of plugins are enabled, if we want additional plugins we should open an new issue for them.   For example, enabling tide should be its own issue, and NOT a "good first issue".
